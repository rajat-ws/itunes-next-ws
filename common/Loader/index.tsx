import { Space, Spin } from "antd";
import React from "react";
import { AlignCenter } from "../styled";

const Loader: React.FC = () => (
  <AlignCenter>
    <Space size="middle">
      <Spin size="large" />
    </Space>
  </AlignCenter>
);

export default Loader;
