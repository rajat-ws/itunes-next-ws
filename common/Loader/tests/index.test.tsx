import { render } from "@app/utils/testUtils";
import Loader from "../index";

describe("<Loader />", () => {
  it("should render and match the snapshot", () => {
    const { baseElement } = render(<Loader />);
    expect(baseElement).toMatchSnapshot();
  });

  it("should render AlignCenter", () => {
    const { getByTestId } = render(<Loader />);
    expect(getByTestId("align-center")).toBeInTheDocument();
  });
});
