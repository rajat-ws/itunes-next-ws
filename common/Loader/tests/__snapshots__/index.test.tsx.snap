// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="Loader__AlignCenter-sc-1fl5bd6-0 gLYoBj"
      data-testid="align-center"
    >
      <div
        class="ant-space ant-space-horizontal ant-space-align-center"
      >
        <div
          class="ant-space-item"
        >
          <div
            aria-busy="true"
            aria-live="polite"
            class="ant-spin ant-spin-lg ant-spin-spinning"
          >
            <span
              class="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
